core/src/main/scala/org/genivi/sota/core/VehicleUpdatesResource.scala

```diff
@@ -154,28 +155,34 @@ class VehicleUpdatesResource(db : Database, resolverClient: ExternalResolverClie
   /**
     * The web app PUT the status of the given ([[UpdateSpec]], VIN) to [[UpdateStatus.Canceled]]
     */
-  def cancelUpdate(vin: Vehicle.Vin, uuid: Refined[String, Uuid]): Route = {
-    val response = db.run(UpdateSpecs.cancelUpdate(vin, uuid)).map {
-      i: Int => i match {
-          case 0 => HttpResponse(StatusCodes.BadRequest)
-          case _ => HttpResponse(StatusCodes.NoContent)
-        }
+  def cancelUpdate(ns: Namespace, vin: Vehicle.Vin, uuid: Refined[String, Uuid]): Route = {
```

Comment:
imo having this kind of logic (0/_) in the "controller" is not so nice. I think the `cancelUpdate` method should return a meaningful value (failed future, try, whatever) and the controller should just convert that to a proper response, at most.